packages/protocol/contracts/common/Accounts.sol

```diff
@@ -0,0 +1,483 @@
+   * @notice Returns the attestation signer for the specified account.
+   * @param account The address of the account.
+   * @return The address with which the account can sign attestations.
+   */
+  function getAttestationSignerFromAccount(address account) public view returns (address) {
+    require(isAccount(account));
+    address attestor = accounts[account].signers.attesting;
+    return attestor == address(0) ? account : attestor;
+  }
+
+    /**
+   * @notice Authorizes voting or validating power of `msg.sender`'s account to another address.
+   * @param current The address to authorize.
+   * @param previous The previous authorized address.
+   * @param v The recovery id of the incoming ECDSA signature.
+   * @param r Output value r of the ECDSA signature.
+   * @param s Output value s of the ECDSA signature.
+   * @dev Fails if the address is already authorized or is an account.
+   * @dev v, r, s constitute `current`'s signature on `msg.sender`.
+   */
+  function authorize(
+    address current,
+    address previous,
+    uint8 v,
+    bytes32 r,
+    bytes32 s
+  )
+    private
+  {
+    require(isAccount(msg.sender) && isNotAccount(current) && isNotAuthorized(current));
+
+    address signer = Signatures.getSignerOfAddress(msg.sender, v, r, s);
+    require(signer == current);
+
+    authorizedBy[previous] = address(0);
+    authorizedBy[current] = msg.sender;
+  }
+
+  /**
+   * @notice Check if an account already exists.
+   * @param account The address of the account
+   * @return Returns `true` if account exists. Returns `false` otherwise.
+   */
+  function isAccount(address account) public view returns (bool) {
+    return (accounts[account].exists);
+  }
+
+  /**
+   * @notice Check if an account already exists.
+   * @param account The address of the account
+   * @return Returns `false` if account exists. Returns `true` otherwise.
+   */
+  function isNotAccount(address account) internal view returns (bool) {
+    return (!accounts[account].exists);
+  }
+
+    /**
+   * @notice Check if an address has been authorized by an account for voting or validating.
+   * @param account The possibly authorized address.
+   * @return Returns `true` if authorized. Returns `false` otherwise.
+   */
+  function isAuthorized(address account) external view returns (bool) {
+    return (authorizedBy[account] != address(0));
+  }
+
+  /**
+   * @notice Check if an address has been authorized by an account for voting or validating.
+   * @param account The possibly authorized address.
+   * @return Returns `false` if authorized. Returns `true` otherwise.
+   */
+  function isNotAuthorized(address account) internal view returns (bool) {
+    return (authorizedBy[account] == address(0));
+  }
+
+  /**
+   * @notice Setter for the metadata of an account.
+   * @param metadataURL The URL to access the metadata.
+   */
+  function setMetadataURL(string calldata metadataURL) external {
+    accounts[msg.sender].metadataURL = metadataURL;
```

Comment:
Should all of these setters require that `msg.sender` is an account? If so, please add tests that check that the tx reverts if not